x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/IndicesAndAliasesResolverTests.java

```diff
@@ -2237,7 +2242,8 @@ public void testDataStreamNotAuthorizedWhenBackingIndicesAreAuthorizedViaWildcar
         assertThat(authorizedIndices.check("logs-foobar"), is(false));
         DataStream dataStream = metadata.dataStreams().get("logs-foobar");
         assertThat(authorizedIndices.all().get(), not(hasItem(indexName)));
-        assertThat(authorizedIndices.check(indexName), is(false));
+        // request pattern is subset of the authorized pattern
+        assertThat(authorizedIndices.check(indexName), is(true));
```

Comment:
I see we actually changing the assertion here. This is what I was suggesting above.